lib/wasi/src/state.rs

```diff
@@ -3,24 +3,30 @@
 //     file_like::{FileLike, Metadata};
 // };
 use crate::syscalls::types::*;
-use generational_arena::{Arena, Index as Inode};
+use generational_arena::Arena;
+pub use generational_arena::Index as Inode;
 use hashbrown::hash_map::{Entry, HashMap};
 use std::{
+    borrow::Borrow,
     cell::Cell,
     fs,
     io::{self, Read, Seek, Write},
-    path::PathBuf,
+    path::{Path, PathBuf},
     time::SystemTime,
 };
 use wasmer_runtime_core::debug;
 
-pub const MAX_SYMLINKS: usize = 100;
+pub const MAX_SYMLINKS: u32 = 128;
```

Comment:
Is this number arbitrary? (or based in other impl?)

Reply:
it's completely arbitrary, which I can explain in a comment.

It's just there to prevent infinite loops.  It seemed like a "big enough" number